x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2035,6 +2035,23 @@ export class RulesClient {
       } catch (e) {
         throw e;
       }
+    }
+
+    let scheduledTaskIdToCreate: string | null = null;
+    if (attributes.scheduledTaskId) {
+      // If scheduledTaskId defined in rule SO, make sure it exists
+      try {
+        await this.taskManager.get(attributes.scheduledTaskId);
+      } catch (err) {
+        scheduledTaskIdToCreate = id;
+      }
+    } else {
+      // If scheduledTaskId doesn't exist in rule SO, set it to rule ID
+      scheduledTaskIdToCreate = id;
+    }
+
+    if (scheduledTaskIdToCreate) {
+      // Schedule the task if it doesn't exist
       const scheduledTask = await this.scheduleRule({
```

Comment:
Nitpick: scheduleTask maybe?

Reply:
Resolved in https://github.com/elastic/kibana/pull/139826/commits/3bd7a3a00798a2936f71d8ab8918bbf1718fad04